Apps/GatewayApi/src/Models/GetDependentResponse.cs

```diff
@@ -0,0 +1,36 @@
+// -------------------------------------------------------------------------
+//  Copyright © 2019 Province of British Columbia
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+// -------------------------------------------------------------------------
+
+namespace HealthGateway.GatewayApi.Models
+{
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Represents a get dependent response.
+    /// </summary>
+    public class GetDependentResponse
+    {
+        /// <summary>
+        /// Gets or sets the delegate id.
+        /// </summary>
+        public string DelegateId { get; set; } = null!;
+
+        /// <summary>
+        /// Gets or sets the dependent records.
+        /// </summary>
+        public IEnumerable<DependentRecord>? DependentRecords { get; set; }
+    }
```

Comment:
I may be missing something, but the story doesn't mention returning a list of results grouped by delegate.  It says the response should include:
- [list of results containing] resource owner, resource delegate, expiry, createddatetime 
    - ordered by created date time 
- page number 
- total count of records

Reply:
You're right, but as there can be multiple delegates per owner, I think it makes sense to return a structured json that contains the hierarchy.

Reply:
A couple of points for discussion:

- The use case doesn't require grouping 
- The query is date/time based and may or may not result in actual grouped data
- The grouping is being done in C# code which I don't like
- I believe the count as returned doesn't match the DB query which means the page size doesn't match what the caller requested.


The last two points should be resolved prior to merging.

Reply:
I guided Ray to do the grouping as it matches closely what I understand PHSA would want to see, we can discuss it and change accordingly if I misunderstood.
I'll work with Ray on moving the grouping (if required to the db layer and fix the count (I agree it would be incorrect).